[test_dispatch_profiler.py] | Sanity tests nightly debug run with LLK asserts | blackhole nightly debug run (Ubuntu 24.04 with LLK asserts) / profiler-tests / L1 accumulate profiler [bh_p150b_civ2_viommu]

## CI Failure Report

<!-- AUTO-TRIAGE-SUMMARY-START -->
**Summary:** [CI] Profiler TT_ASSERT failure in readRiscProfilerResults: riscNumRead == riscEndIndex (BH L1 accumulate profiler)
<!-- AUTO-TRIAGE-SUMMARY-END -->

**Workflow:** Sanity tests nightly debug run with LLK asserts
**Job:** blackhole nightly debug run (Ubuntu 24.04 with LLK asserts) / profiler-tests / L1 accumulate profiler [bh_p150b_civ2_viommu]

### Regression timeline

- **Last passing commit:** `7a29801d6794` — 2026-07-14T03:01:53Z — https://github.com/tenstorrent/tt-metal/actions/runs/29302503247
- **First failing commit:** `746fd1cf5ea4` — 2026-07-21T03:02:28Z — https://github.com/tenstorrent/tt-metal/actions/runs/29797432775/job/88545401263
- **Most recent failing commit:** `df36cd6a7434` — 2026-07-22T03:02:33Z — https://github.com/tenstorrent/tt-metal/actions/runs/29887396291/job/88834824741

### Failing job URLs (last 2 runs)

- https://github.com/tenstorrent/tt-metal/actions/runs/29887396291/job/88834824741
- https://github.com/tenstorrent/tt-metal/actions/runs/29797432775/job/88545401263

### Failing test path(s)

- `tests/ttnn/tracy/test_dispatch_profiler.py::test_with_ops[device_params={'dispatch_core_type': DispatchCoreType.WORKER}]`

### Error excerpt (terminal failure from logs)

**Run 1 (commit `df36cd6a7434`, 2026-07-22):**
```
ERROR tests/ttnn/tracy/test_dispatch_profiler.py::test_with_ops[device_params={'dispatch_core_type': DispatchCoreType.WORKER}] - RuntimeError: TT_ASSERT @ /project/tt_metal/impl/profiler/profiler.cpp:1712: riscNumRead == riscEndIndex
info:
Unexpected risc id, expected 0, read 8. In core 1,2 TENSIX at run 0, index 60012
================== 1 passed, 1 deselected, 1 error in 13.20s ===================
##[error]Process completed with exit code 1.
```

**Run 2 (commit `746fd1cf5ea4`, 2026-07-21):**
```
ERROR tests/ttnn/tracy/test_dispatch_profiler.py::test_with_ops[device_params={'dispatch_core_type': DispatchCoreType.WORKER}] - RuntimeError: TT_ASSERT @ /project/tt_metal/impl/profiler/profiler.cpp:1712: riscNumRead == riscEndIndex
info:
Unexpected risc id, expected 0, read 8. In core 1,2 TENSIX at run 0, index 60012
================== 1 passed, 1 deselected, 1 error in 13.49s ===================
##[error]Process completed with exit code 1.
```

The assertion fires during teardown in `tt::tt_metal::DeviceProfiler::readRiscProfilerResults`, propagating through `processResults` → `ReadMeshDeviceProfilerResults` → `MeshDeviceImpl::close_impl` → `MeshDevice::close()`.

### Reproduction steps

On a Blackhole P150B device with profiler build:

```bash
python -m pytest tests/ttnn/tracy/test_dispatch_profiler.py::test_with_ops \
  -k "dispatch_core_type=DispatchCoreType.WORKER" \
  --timeout=300
```

Or using the workflow's test selection (post-commit profiler tests on `bh_p150b_civ2_viommu` SKU).

### Notes

- The test body **passes** (`PASSED` appears in the summary); the failure is in **teardown** when the device is closed and profiler results are read back.
- The assertion `riscNumRead == riscEndIndex` in `profiler.cpp:1712` indicates a mismatch between expected and actual RISC core IDs when parsing profiler data — specifically `Unexpected risc id, expected 0, read 8. In core 1,2 TENSIX at run 0, index 60012`.
- This suggests a profiler buffer corruption or mismatched RISC ID encoding introduced between `7a29801d6794` (2026-07-14, last passing) and `746fd1cf5ea4` (2026-07-21, first failing).
- **Determinism check:**
  - Run 1 terminal error: `RuntimeError: TT_ASSERT @ /project/tt_metal/impl/profiler/profiler.cpp:1712: riscNumRead == riscEndIndex` — `Unexpected risc id, expected 0, read 8. In core 1,2 TENSIX at run 0, index 60012`
  - Run 2 terminal error: `RuntimeError: TT_ASSERT @ /project/tt_metal/impl/profiler/profiler.cpp:1712: riscNumRead == riscEndIndex` — `Unexpected risc id, expected 0, read 8. In core 1,2 TENSIX at run 0, index 60012`
  - Both runs: identical assertion, identical RISC ID values, identical core coordinates, identical buffer index.

<!-- AUTO-TRIAGE-METADATA-START -->
`Auto-triage-workflow: Sanity tests nightly debug run with LLK asserts`
`Auto-triage-job-name: blackhole nightly debug run (Ubuntu 24.04 with LLK asserts) / profiler-tests / L1 accumulate profiler [bh_p150b_civ2_viommu]`
<!-- AUTO-TRIAGE-METADATA-END -->

<!-- AUTO-TRIAGE-INFO-START -->
## About this issue (automated CI auto-triage)
This issue was opened and assigned by an automated AI system. It posts through a bot token, so notifications may look like they come from a person, but they are automated.

The owner is selected from recent commit history and job-ownership data. It reflects who is *likely able to fix or route* this, not necessarily who caused it, so it can be imperfect.

**If this was assigned to the wrong person, you can:**
- Comment `REASSIGN OWNER: <why this is outside your area / which team or test-area it belongs to>` and the system will automatically re-route it to a better-suited owner.
- Unassign yourself (this stops notifications to you).
- Close the issue if it is not a real or relevant failure.

The assigned owner is reminded about every 3 days until the issue is resolved. Full guide: https://tenstorrent.atlassian.net/wiki/spaces/MI6/pages/2545221692/Auto-Triage+Info
<!-- AUTO-TRIAGE-INFO-END -->

--- a/tt_metal/impl/profiler/profiler.cpp
+++ b/tt_metal/impl/profiler/profiler.cpp
@@ -2588,7 +2588,9 @@ void DeviceProfiler::pushTracyDeviceResults(
 #if defined(TRACY_ENABLE)
     ZoneScoped;
     if (!getDeviceProfilerState(context_id) ||
-        MetalContext::instance(context_id).rtoptions().get_profiler_disable_push_to_tracy()) {
+        MetalContext::instance(context_id).rtoptions().get_profiler_disable_push_to_tracy() ||
+        // TODO: Quasar is CSV-only for now
+        device_arch == tt::ARCH::QUASAR) {
         return;
     }
 

--- a/tt_metal/tools/profiler/kernel_profiler.hpp
+++ b/tt_metal/tools/profiler/kernel_profiler.hpp
@@ -401,7 +401,9 @@ __attribute__((noinline)) void finish_profiler(bool do_accumulate = DO_ACCUMULAT
                 }
                 // Preserve the upper (trace ID) bits of ID_LH; stamp core + risc id.
                 profiler_data_buffer[riscID].data[ID_LH] =
-                    (profiler_data_buffer[riscID].data[ID_LH] & 0x7FFF800) | (((core_flat_id & 0xFF) << 3) | riscID);
+                    (profiler_data_buffer[riscID].data[ID_LH] & PROFILER_ID_TRACE_FIELD_MASK) |
+                    (((core_flat_id & PROFILER_ID_FLAT_MASK) << PROFILER_ID_FLAT_SHIFT) |
+                     (riscID & PROFILER_ID_RISC_MASK));
 
                 int hostIndex = kernel_profiler::HOST_BUFFER_END_INDEX_BR_ER + riscID;
                 int deviceIndex = kernel_profiler::DEVICE_BUFFER_END_INDEX_BR_ER + riscID;

